Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -48,41 +49,56 @@ namespace AzToolsFramework
+        for (auto propertyWidgetIter = m_widgetToPropertyHandler.begin(), endIter = m_widgetToPropertyHandler.end();
+             propertyWidgetIter != endIter; ++propertyWidgetIter)
+        {
+            m_columnLayout->removeWidget(propertyWidgetIter->first);
+        }
+        m_widgetToPropertyHandler.clear();
+
+        DestroyLayoutContents(m_columnLayout);
+        DestroyLayoutContents(m_childRowLayout);
         m_domOrderedChildren.clear();
     }
 
     void DPERowWidget::AddChildFromDomValue(const AZ::Dom::Value& childValue, int domIndex)
     {
-        auto childType = childValue.GetNodeName().GetStringView();
+        auto childType = childValue.GetNodeName();
 
         // create a child widget from the given DOM value and add it to the correct layout
         QBoxLayout* layoutForWidget = nullptr;
         QWidget* addedWidget = nullptr;
-        if (childType == AZ::DocumentPropertyEditor::Nodes::Row::Name)
+        if (childType == AZ::Dpe::GetNodeName<AZ::Dpe::Nodes::Row>())
         {
             // if it's a row, recursively populate the children from the DOM array in the passed value
             auto newRow = new DPERowWidget(this);
-            newRow->SetChildrenFromDomArray(childValue);
+            newRow->SetValueFromDom(childValue);
             addedWidget = newRow;
             layoutForWidget = m_childRowLayout;
         }
-        else if (childType == AZ::DocumentPropertyEditor::Nodes::Label::Name)
+        else if (childType == AZ::Dpe::GetNodeName<AZ::Dpe::Nodes::Label>())
         {
             auto labelString = childValue[AZ::DocumentPropertyEditor::Nodes::Label::Value.GetName()].GetString();
             addedWidget = new QLabel(QString::fromUtf8(labelString.data(), static_cast<int>(labelString.size())), this);
             layoutForWidget = m_columnLayout;
         }
-        else if (childType == AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Name)
+        else if (childType == AZ::Dpe::GetNodeName<AZ::Dpe::Nodes::PropertyEditor>())
         {
-            // todo: use line edits with the dom text for now. Replace this with actual handler widgets, once they are done
-            AZ::Dom::JsonBackend<AZ::Dom::Json::ParseFlags::ParseComments, AZ::Dom::Json::OutputFormatting::MinifiedJson> jsonBackend;
-            AZStd::string stringBuffer;
-            AZ::Dom::Utils::ValueToSerializedString(
-                jsonBackend, childValue[AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Value.GetName()], stringBuffer);
+            auto dpeSystem = AZ::Interface<AzToolsFramework::PropertyEditorToolsSystemInterface>::Get();
+            auto handlerId = dpeSystem->GetPropertyHandlerForNode(childValue);
 
-            addedWidget = new QLineEdit(QString::fromUtf8(stringBuffer.data(), static_cast<int>(stringBuffer.size())), this);
+            // <apm> this shouldn't happen once everything is implemented, but bail for now on a null handlerId
```

Comment:
We'll probably want to log a warning or something if this happens, an Adapter could always accidentally ask for a non-existent handler.